app/services/katello/pulp3/docker_manifest.rb

```diff
@@ -0,0 +1,36 @@
+module Katello
+  module Pulp3
+    class DockerManifest < PulpContentUnit
+      include LazyAccessor
+
+      def self.content_api
+        PulpDockerClient::ContentManifestsApi.new(Katello::Pulp3::Repository::Docker.api_client(SmartProxy.pulp_master!))
+      end
+
+      def self.ids_for_repository(repo_id)
+        repo = Katello::Pulp3::Repository::Docker.new(Katello::Repository.find(repo_id), SmartProxy.pulp_master)
+        repo_content_list = repo.content_list
+        repo_content_list.map { |content| content.try(:_href) }
+      end
+
+      def self.content_unit_list(page_opts)
+        page_opts[:media_type] = "application/vnd.docker.distribution.manifest.v1+json"
+        data_v1 = self.content_api.list(page_opts)
+        page_opts[:media_type] = "application/vnd.docker.distribution.manifest.v2+json"
+        data_v2 = self.content_api.list(page_opts)
+
+        filtered = {}
+        filtered["count"] = data_v1.count + data_v2.count
+        filtered["results"] = data_v1.results + data_v2.results
+        filtered
+      end
```

Comment:
Does this work properly with pagination?  i would think it would make a lot of extra requests potentially if it worked at all for large repos.  Is it possible to pass media_type as an array or do some other type of filtering so that we can just make a single request?

Reply:
Filtering by both media types would definitely be preferable, I'm seeing if that's possible in Pulp 3.

Reply:
I've added a story for it here https://pulp.plan.io/issues/5157

Reply:
Thought about this a bit more and i think its fine for now

Reply:
i opened https://projects.theforeman.org/issues/27425 to track it

Reply:
ok, and I think the solution will be the filtering that the Pulp team is adding.